File: lib/src/main/java/vserver/route/FixedSubPath.java
package vserver.route;

import vserver.SubPath;
import vserver.RoutingContext;

public class FixedSubPath implements SubPath {
    private final SubPath next;
    private final String route;

    public FixedSubPath(SubPath next, String route) {
        this.next = next;
        this.route = route;
    }

    @Override
    public SubPath next() {
        return next;
    }

    @Override
    public boolean match(String route) {
        return this.route.equals(route);
    }

    @Override
    public boolean currentSame(SubPath r) {
        return r instanceof FixedSubPath && ((FixedSubPath) r).route.equals(route);
    }

    @Override
    public void fill(RoutingContext ctx, String route) {
    }
}

File: lib/src/main/kotlin/io/vproxy/lib/http/route/FixedSubPath.kt
package io.vproxy.lib.http.route

import vproxy.lib.http.RoutingContext

class FixedSubPath(private val next: SubPath?, private val route: String) : SubPath {
  override fun next(): SubPath? {
    return next
  }

  override fun match(route: String): Boolean {
    return this.route == route
  }

  override fun currentSame(r: SubPath): Boolean {
    return r is FixedSubPath && r.route == route
  }

  override fun fill(ctx: RoutingContext, route: String) {}

  override fun toString(): String {
    return "/$route"
  }
}
